### ISSUE TITLE
[STORY]: Update 'Status' to 'Mark as Interested' feature

### ISSUE THREAD

**Issue by ekhoffman** [usdigitalresponse/usdr-gost#2960]
### Full User Story

As an organization, we would like to foster more collaboration between our users to allow for the sharing of knowledge and resources to empower them to more efficiently and effectively apply for the grants discovered on our platform.

As a grantseeker, I want to be able to easily share grants I’ve discovered with my team.

As a grantseeker, I want to be able to easily find collaborators who may be able to help me when preparing and applying for grants.

As a grantseeker, I want to be able to quickly see what grants my team members are interested in.

As an admin, I want to be able to easily surface and resolve issues and blockers my team is running into.

### Why is this issue important?

We have observed that the 'Status' feature is very underused as it currently stands. We've additionally received feedback from our partners that they believe one of the most valuable uses for our tool would be to aid in building collaborative networks. Our goal in shifting from 'Status' to 'Mark as Interested' is to help increase usage and visibility at the individual level to help inspire collaboration.

### Action Plan

- [ ] Initial design drafted
- [ ] Initial acceptance criteria drafted
- [ ] High level technical approach & feature flag use determined
- [ ] Designs & Acceptance criteria finalized
- [ ] Eng task breakout complete
- [ ] Assessment of whether partner training & communication actions are needed
- [ ] Acceptance criteria testing completed & passed in staging
- [ ] Feature flags enabled in production
- [ ] Feature flags are removed once the feature is deemed stable

### Engineering Tasks

Add links to engineering tasks

### Acceptance Criteria

**Designs:** https://www.figma.com/file/L2gfVMBH0x1daOlOe50sJl/Mark-as-Interested-%2F-Share-to-Team?type=design&node-id=532-746&mode=design&t=OVNpgdt5VzEi9brI-4

**Grant Details Page**
1. Remove the current 'Status' section including the drop-down box and 'Submit' button.
2. Add 'Interested' section that includes the following:
    - 'Interested' section header
    - 'Mark as Interested' button
        - The button will have two states, primary and secondary, and appear differently based on the current state.
            - If the User has not yet marked interest, they will see the primary button state.
            - If Primary, button text will be "Mark as Interested."
            - If the User has marked interested, they will see the secondary button state.
            - If Secondary, button text will be "Marked as Interested."
3. The button will also have a descriptive header that changes based on the button state.
    - If Primary, the descriptive header will say "Let your network know you're interested."
    - If Secondary, the descriptive header will say "We'll let your network know!"
4. Upon clicking the button in the Primary State, it will change to the Secondary State and the following new elements will appear:
    - Container Shading
    - Edit Button in top right corner of container
    - Current User Summary Card containing the following information: 
      Current User Icon, Current User Name, Current User Team, Current User Email (with a copy button next to it)
    - Interest Note Entry Box (with the example text "Share any barriers to applying".
    - Interest Note Descriptive Text ("e.g. I need cost matching, I can support").
5. Upon submitting an Interest Note, the text entry box will transition to display the values entered as Read-Only.
6. The user will be able to click the Edit button when they've previously marked interest and do the following:
    - Un-mark Interest
    - Edit Interest Note
7. Below this we will display a feed of all users within the Current User's organization who have marked interest. The feed will include the following:
    - User Summary Cards for each User who has marked Interest containing the following information: 
      - User Icon, User Name, User Team, User Email (with a copy button next to it), User Interest Note (if entered), Date of Interest Marked
    - The display of these summary cards will be display the most recent interest marker and then descend from there.

**My Grants**
1. Update Interested table to work with new 'Mark as Interested' value.
  - Only display values of those that were marked by the Current User.

**Dashboard**
1. Update 'Recent Activity' feed to work with the new 'Mark as Interested' value.
    - When Individual team members mark interest, they should appear in the feed with "{{User Name}} is interested in {{Grant Name}}."
2. Update the 'Upcoming Closing Dates' feed to work with the the new 'Mark as Interested' value.
    - 

Expected behavior includes…

**Comment by ClaireValdivia** [comment]
Note - one outstanding requirement here on Dashboard for csv export - curious to know about complexity of csv export

**Comment by TylerHendrickson** [comment]
@ClaireValdivia Clarifying question: on the Grant Details Page views, are we only showing followers and notes to other users in the same organization? Are there any other restrictions (e.g. team) that need to be taken into consideration?

**Comment by ClaireValdivia** [comment]
@TylerHendrickson  You can see followers and notes across the entire organization, no restrictions by Team! You can't see anything from a different organization.

**Comment by TylerHendrickson** [comment]
@ClaireValdivia / @caitlinwinner – re

> - Grant Activity Summary
    - This will display below the Follow button and include both Follower information AND a count of notes. (If both have values, they will be separated by a bullet point.)
    - The Follower information will display the following:
       - "Followed by {{Follower Name}} and {{Count Followers - 1}} others" _(Note: If there are notes related to the grant, this will be followed by a bullet point and count of notes. If there are no followers we will suppress this link fully.)_

What determines `{{ Follower Name }}`? Is oldest follower (who is still following), the most recent follower, a random follower, or some other determining factor? From a technical standpoint, any of those implementations should all be about the same LoE. From a UX perspective, I'm wondering if the first person to follow a grant would want to be "memorialized" on that grant henceforth (with no way to take their name down other than unfollowing), and potentially the same situation (although more impacted by follower activity) for the last person. Is there a reason to (not) favor consistently showing the same name on a grant?

**Comment by TylerHendrickson** [comment]
@ClaireValdivia / @caitlinwinner – re

> - The Grant Followers modal will include the following:
>   - A header of 'Grant Followers'
>   - A list of all followers sorted alphabetically in ascending order by name.

Just seeking confirmation that this is the sort order we want for the modal. The designs also seem to imply that the sort order is based on "most-recently followed first".

**Comment by ClaireValdivia** [comment]
I don't have a strong opinion here, but Ty's recommendation of most recent follower for the linked text, and displaying the followers in order of most-recently-followed (rather than alphabetical) makes sense to me. @caitlinwinner @Samcbehrens curious for your thoughts!

**Comment by samserif23** [comment]
I agree that the recency approach seems more helpful than an alphabetical approach.

**Comment by TylerHendrickson** [comment]
@ClaireValdivia / @caitlinwinner / @samserif23 Thanks for confirming the display order for followers!

Separate request: can someone clarify when the "Leave a note with..." textbox is supposed to display? I'm unclear from the current designs whether it should show up to a the user who is **not** currently following the grant. And if the textbox does appear for non-followers, does leaving a note automatically make the user a follower?

**Comment by ClaireValdivia** [comment]
@TylerHendrickson I just updated the issue to reflect the display order.

The "leave a note with" textbox should display for all users who have not submitted a note. Once they submit a note, their not displays at the top instead of the textbox; if they delete their note, it goes back to displaying the text box.
